zapgrpc/zapgrpc.go

```diff
@@ -48,38 +51,95 @@ func WithDebug() Option {
 // By default, Loggers print at zap's InfoLevel.
 func NewLogger(l *zap.Logger, options ...Option) *Logger {
 	logger := &Logger{
-		log:    l.Sugar(),
-		fatal:  (*zap.SugaredLogger).Fatal,
-		fatalf: (*zap.SugaredLogger).Fatalf,
-		print:  (*zap.SugaredLogger).Info,
-		printf: (*zap.SugaredLogger).Infof,
+		log:      l.Sugar(),
+		info:     (*zap.SugaredLogger).Info,
```

Comment:
is there a reason why we have to keep function pointers for each level, instead of just doing `l.sugar.Info(..)` etc?

Reply:
@prashantv no reason. I was just trying to maintain the programming style that was previously present.

Reply:
@prashantv this is for the `WithDebug` function, which changes the zap function used for the `Print` functions.

Reply:
Ah, I see. In that case, aren't the only functions we need to do this for `print` and `printf`? The rest should just be normal function calls on the Logger.

Reply:
That would work too, either/or, I was just being consistent, especially “back in the day” before this PR when there were only a few functions :-)